File: atsa/src/main/java/jp/kaiz/atsassistmod/util/DataType.java
package jp.kaiz.atsassistmod.util;

/** Local copy of RTM's modelpack DataType (used by IFTTT data-map rules). */
public enum DataType {
    BOOLEAN("bool"),
    INT("int"),
    HEX("hex"),
    DOUBLE("double"),
    STRING("string"),
    VEC("vec");

    public final String key;

    DataType(String key) {
        this.key = key;
    }
}

File: atsa/src/main/kotlin/jp/kaiz/atsassistmod/util/DataType.kt
// SPDX-License-Identifier: LGPL-3.0-or-later
// Copyright © 2026 mirukuneko and RealTrainModRenewed contributors
package jp.kaiz.atsassistmod.util

/** Local copy of RTM's modelpack DataType (used by IFTTT data-map rules). */
enum class DataType(@JvmField val key: String) {
    BOOLEAN("bool"),
    INT("int"),
    HEX("hex"),
    DOUBLE("double"),
    STRING("string"),
    VEC("vec"),
}
